src/App.js

```diff
@@ -7,46 +7,45 @@ import Container from './components/app-container';
 import LandingPage from './routes/landingPage';
 
 export default class App extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-          isLoggedIn: JSON.parse(localStorage.getItem('isLoggedIn'))
-        }
-        this.logIn = this.logIn.bind(this);
-        this.logOut = this.logOut.bind(this);
-    }
-
-    componentDidMount() {
-        const userPresets = localStorage.getItem('userPresets');
-        const customSpaces = localStorage.getItem('customSpaces');
-        if (!userPresets) {
-            localStorage.setItem('userPresets', JSON.stringify({
-              "data": []
-            }))
-        }
-
-        if (!customSpaces) {
+
+  constructor(props) {
+      super(props);
+      this.state = {
+        isLoggedIn: JSON.parse(localStorage.getItem('isLoggedIn'))
+      }
+      this.logIn = this.logIn.bind(this);
+      this.logOut = this.logOut.bind(this);
+  }
+
+  /**
+   * Local Storage must be set up before rendering the app because some
+   * filtered study space information is distributed to various pages.
+   */
+  componentWillMount() {
```

Comment:
componentWillMount is dangerous. Dont know how this came to be, but it should be componentDidMount(). Check week 7 code again

Reply:
ok fixed.